lib/licensed/sources/gradle.rb

```diff
@@ -122,61 +79,132 @@ def configurations
         end
       end
 
-      def self.add_gradle_license_report_plugins_block(gradle_build_file)
+      # Prefixes the gradle command with the project name for multi-build projects.
+      def format_command(command)
+        if config.source_path != config.pwd
+          project = File.basename(config.source_path)
+        end
+        project.nil? ? command : "#{project}:#{command}"
+      end
+
+      # Returns a key to uniquely identify a name and version in the obtained CSV content
+      def csv_key(name:, version:)
+        "#{name}-#{version}"
+      end
+
+      def package_url(name:, version:)
+        # load and memoize the license report CSV
+        @urls ||= load_csv
 
-        if gradle_build_file.include? "plugins"
-          gradle_build_file.gsub(/(?<=plugins)\s+{/, " { id 'com.github.jk1.dependency-license-report' version '1.16'")
-        else
+        # uniquely identify a name and version in the obtained CSV content
+        @urls["#{name}-#{version}"]
+      end
 
-          gradle_build_file = " plugins { id 'com.github.jk1.dependency-license-report' version '1.16' }" + gradle_build_file
+      def load_csv
+        begin
+        # create the CSV file including dependency license urls using the gradle plugin
+          gradle_runner.run("generateLicenseReport")
+
+          # parse the CSV report for dependency license urls
+          gradle_licenses_dir = File.join(config.root, GRADLE_LICENSES_PATH)
+          CSV.foreach(File.join(gradle_licenses_dir, GRADLE_LICENSES_CSV_NAME), headers: true).each_with_object({}) do |row, hsh|
+            name, _, version = row["artifact"].rpartition(":")
+            key = csv_key(name: name, version: version)
+            hsh[key] = row["moduleLicenseUrl"]
+          end
+        ensure
+          FileUtils.rm_rf(gradle_licenses_dir)
```

Comment:
I'm guessing the error being thrown in CI is due to `gradle_runner.run` raising an error before `gradle_licenses_dir` is set, sorry for not accounting for that in my suggested code.  moving `gradle_licenses_dir = File.join(...)` above `gradle_runner.run("generateLicenseReport")` should fix that